lib/environment.js

```diff
@@ -94,6 +94,75 @@ class BasicMDP extends Environment {
 	}
 }
 
+//Implement time inconsistent env based on MDP implementation for now
+class TimeInconsistentEnv1 extends Environment {
+	constructor(config) {
+		super(config);
+		this.states = [0, 1, 2];
+		this.pos_array = [{ x: 190, y: 60 }, { x: 80, y: 160 },  { x: 300, y: 160 }];
+		this.delayed_dispense = 6; //Time at which delay state will return a non-zero reward
+		this.delayed_reward = 1000;
+		this.instant_reward = 4;
+		let numActions = 3; //instant and delayed actions
+		this.actions = new Array(numActions);
+		this.current = this.states[0];
+		this.delayed_counter = 0; //if this.current = delayed : counter++ else counter = 0
+	}
+
+	perform(action) {
+		this.current = this.states[action];
+		if (this.current == 2) {
+			this.delayed_counter++;
+
+			if (this.delayed_counter == this.delayed_dispense) {
+				this.reward = this.delayed_reward;
+				this.delayed_counter = 0;
+			} else {
+				this.reward = 0;
+			}
+
+		} else {
+			this.delayed_counter = 0;
+			(this.current == 1) ? this.reward = this.instant_reward : this.reward = 0;
+		}
+	}
+
+	generatePercept() {
+		return {
+			obs: this.current,
+			rew: this.reward,
+		};
+	}
+
+	getState() {
+		return {
+			current : this.current,
+			count : this.delayed_counter,
+			reward : this.reward,
+
+		};
+	}
+
+	save() {
+		this.state = {
+			current: this.current,
+			counter: this.delayed_counter,
+			reward: this.reward,
+		};
+	}
+
+	load() {
+		Util.assert(this.state, 'No saved state to load!');
+		this.current = this.state.current;
+		this.delayed_counter = this.state.counter;
+		this.reward = this.state.reward;
+	}
+
+	conditionalDistribution(e) {
```

Comment:
returning 1 here only works because we're using AImu, so it's a hack. either calculate the actual probability or leave a note here that this isn't actually correct in general